nvflare/client/api.py

```diff
@@ -49,46 +51,65 @@ def init(config: Union[str, Dict] = f"config/{CONFIG_EXCHANGE}", rank: Optional[
     else:
         raise ValueError("config should be either a string or dictionary.")
 
-    mdx = None
+    dx = None
     if rank == "0":
-        if client_config.get_exchange_format() == ModelExchangeFormat.PYTORCH:
+        if client_config.get_exchange_format() == ExchangeFormat.PYTORCH:
             tensor_decomposer, ok = optional_import(module="nvflare.app_opt.pt.decomposers", name="TensorDecomposer")
             if ok:
                 fobs.register(tensor_decomposer)
             else:
-                raise RuntimeError(f"Can't import TensorDecomposer for format: {ModelExchangeFormat.PYTORCH}")
+                raise RuntimeError(f"Can't import TensorDecomposer for format: {ExchangeFormat.PYTORCH}")
 
         # TODO: make Pipe configurable
-        mdx = FilePipeModelExchanger(data_exchange_path=client_config.get_exchange_path())
+        dx = FilePipeDataExchanger(
+            supported_topics=client_config.get_supported_topics(),
+            data_exchange_path=client_config.get_exchange_path(),
+            pipe_name=client_config.get_pipe_name(),
+        )
 
-    PROCESS_MODEL_REGISTRY[pid] = ModelRegistry(client_config, rank, mdx)
+    PROCESS_MODEL_REGISTRY[pid] = ModelRegistry(client_config, rank, dx)
 
 
-def _get_model_registry() -> Optional[ModelRegistry]:
+def _get_model_registry() -> ModelRegistry:
```

Comment:
why do you need a dict based on pid? Why can't the model registry be just a global singleton?